### ISSUE TITLE
Ratings with Review

### ISSUE THREAD

**Issue by sujaypawar** [gocodebox/lifterlms#420]
May I suggest to add ratings functionality while adding a review?

**Comment by thomasplevy** [comment]
Yes, we'll be adding ratings in the future as we improve the reviews functionality.

**Comment by actual-saurabh** [comment]
Currently, Reviews are a custom post type. The suggested plan is to migrate reviews from post types into comment types and review titles and ratings into comment meta in a manner similar to how WooCommerce does it: https://github.com/woocommerce/woocommerce/blob/99ec658684284443e157326f5d52cecb554b07b5/includes/class-wc-comments.php

In addition creating a comment type abstraction and the builtin nested nature of comments can allow us to build the following functionality apart from reviews and ratings:

 * **Questions and Answers** similar to https://wordpress.org/plugins/yith-woocommerce-questions-and-answers/ and https://www.skyverge.com/product/woocommerce-product-reviews-pro/. This is inspired from Amazon's Questions. The question would be a separate comment type and the answer(s) would be another. The relationship between questions and answers can be defined using parent-child relationship.
     * This can be extended into a full-fledged **contextual help system** for students to ask questions about a lesson, quiz, question, assignment, etc.
 * **Student Notes** Flat comment types can be used to take private/public notes by a student.
     * WooCommerce also uses this for order notes which is something we can implement with this, as well.
 * **Discussions** Existing comments can be used to power forum like discussion system on each course element.

Any (all) of the above can be easily enhanced with something like http://annotatorjs.org/ to add inline annotations that can be additional explanations added by the instructor, for QnAs to be specific to a piece of text (or with video and audio playback tracking, to a specific playback time).

With User Segmentation in place (#604), this could be used to create really granular and specific visibility for individual users, roles and segments (#610).

**Comment by actual-saurabh** [comment]
https://rateyo.fundoocode.ninja/
